R/run-nmtran.R

```diff
@@ -0,0 +1,223 @@
+#' Interface for running `NM-TRAN` on model objects
+#'
+#' Provides functions to run `NM-TRAN` on model objects for validation and to
+#' generate `NM-TRAN` datasets (`FDATA`).
+#'
+#' @details
+#' `NM-TRAN` is a preprocessor for `NONMEM` that translates user-specified
+#' control stream data and instructions into a form executable by `NONMEM`.
+#'
+#' `run_nmtran()` allows users to test their models ahead of submission to ensure
+#' correct coding.
+#'
+#' @section Supported `bbi_args`:
+#' `run_nmtran()` supports passing the following raw `NMFE` options through to `NM-TRAN`
+#' via the familiar `.bbi_args` argument.
+#'  - **`maxlim`**: Set the maximum values for the buffers used by `NONMEM`
+#'  (if 0, don't pass `-maxlim` to nmfe) (default is 2)
+#'    - If `maxlim = 3`, it is preferred to also set `tprdefault = TRUE`, but
+#'    _not_ `prdefault`, as `NM-TRAN`'s optional resizing of the `PREDPP` size
+#'    parameter `MAXRECID` may conflict with the `-prdefault` option.
+#'  - **`prdefault`**: If `TRUE`, do not recompile any routines other than `FSUBS`
+#'  - **`tprdefault`**: If `TRUE`, test if is okay to do `-prdefault`
+#'
+#' @param .mod A `bbr` model object.
+#' @param .bbi_args A named list specifying arguments to pass to `NM-TRAN`.
+#'   Similar to the `.bbi_args` argument defined in [submit_model()], though here
+#'   only `prdefault`, `tprdefault`, and `maxlim` flags are passed to `NM-TRAN`.
+#'   See details.
+#' @param .config_path Path to a bbi configuration file. If `NULL`, the default,
+#'   will attempt to use a `bbi.yaml` in the same directory as the model.
+#' @param nmtran_exe Path to an `NM-TRAN` executable. If `NULL`, will look for a
+#'   `bbi.yaml` file in the same directory as the model.
+#' @param delete_on_exit Logical. If `FALSE`, don't delete the temporary folder
+#'   containing the `NM-TRAN` run, which will be stored in the current working
+#'   directory.
```

Comment:
Is there a need to expose these yet?

For `nmtran_exe`: if we do want to allow specifying the version, it seems better to allow the user to specify a `nm_version` argument that's consistent with what's done through `.bbi_args` for `submit_model`. That'd require the user to run `bbi_init` or pass `.config_path`, but I think that's fine.

Reply:
Im open to adding support for `nm_version` (perhaps in addition to `nmtran_exe`), but I believe @seth127 requested I add an `nmtran_exe` arg a while back. Im struggling to recall if there was a notable reason for exposing this argument, but I do like mimicking `submit_model` as closely as possible.

With respect to `delete_on_exit`, yes that is an argument we definitely want to keep exposed. Have shown the feature to @kylebaron, and the ability to manually inspect `FDATA` and some other files was helpful. A later patch could read this information in and store it as part of what's returned (not talking about `FDATA` here), though I think that's a bit beyond this PR

Reply:
> Im struggling to recall if there was a notable reason for exposing this argument, but I do like mimicking `submit_model` as closely as possible.

It can be exposed later if that need surfaces, so I'd default to dropping it.

> re: delete_on_exit

I'd suggest renaming it `clean` for consistency with `withr::local_tempdir` (which [I think would be good to use underneath](https://github.com/metrumresearchgroup/bbr/pull/705#discussion_r1686985582).)

> A later patch could read this information in and store it as part of what's returned (not talking about `FDATA` here), though I think that's a bit beyond this PR

Hmm, I don't know if I understand what you're thinking, but that doesn't seem necessary to me.  Anyway, agreed that that's beyond the scope of this PR.

Reply:
>Hmm, I don't know if I understand what you're thinking

I  just meant reading in/organizing any of this information. I'm not suggesting we should read in/attempt to organize anything from these files at any point, just that there is information there beyond `FDATA`.
![image](https://github.com/user-attachments/assets/19a1c402-15e2-40f2-b95f-85e187ba96b0)